### ISSUE TITLE
Merchants want to control purchase option order in Overlays and specify which one is pre-selected

### ISSUE THREAD

**Issue by ksmoore125** [laterpay/laterpay-wordpress-plugin#1136]
# User Story
As a LaterPay merchant, I would like to control the order that items appear in the purchase overlay so that I can ensure my pricing overlay is cohesive and does not add any friction to the end user experience.

# Acceptance Criteria
Update the `LaterPay Content` Section
- It should expand and collapse like the `Global Default Price` Section
- Styles should match the `Global Default Price` Section

Allow the merchant to customize the order of the items in the purchase overlay by:
- Create a `Customize Overlay Section`
  - Should expand & collapse & be styled like the `Global Default Price` Section
  - It should allow the merchant to select the order that the purchase options are displayed in the overlay
    - Default Order `selected by default`
    - Ascending
    - Descending
  - It should allow the merchant to select which purchase option is selected by default
_In the options below, if they choose Time Pass or Subscription and there are multiple Time Passes or Subscriptions available, we should pre-select the first one._
For example: If they choose `By Price (Descending) - Most expensive option listed first` and `Time Pass` the pre-selected option would be the most expensive time pass.
    - The First Option `selected by default`
    - Single Purchase
    - Time Pass
    - Subscription

# Copy

English | German | Notes
-- | -- | --
**In what order should the purchase options be displayed?** | **In welcher Reihenfolge sollen die Kaufoptionen angezeigt werden?** | 
**Default Order** - Single Purchase, Time Pass, the Subscription. | **Standardbestellung** - Einzelkauf, Zeitpässe, Abonnement.| 
If multiple Time Passes or Subscriptions, they will be shown in the order they were created. | Wenn mehrere Zeitpässe oder Abonnements bestehen, werden sie in der Reihenfolge ihrer Erstellung angezeigt.| 
**By Price (Ascending)** - Least expensive option listed first | **Nach Preis (Aufsteigend)** - Günstigste Option zuerst aufgeführt| 
**By Price (Descending)** - Most expensive option listed first | **Nach Preis (Absteigend)** - Teuerste Option zuerst aufgeführt | 
**Which purchase option should be selected by default?** | Welche Kaufoption soll als Standard eingestellt werden? | 
First Option (as defined above) |Erste Option (wie oben definiert) | 
Single Purchase |Einzelkauf | 
Time Pass |Zeitpass | 
Subscription |Abo | 
If there are multiple Time Passes available, the first option (as defined above) will be selected by default. | Wenn mehrere Zeitpässe verfügbar sind, wird automatisch die erste Option (wie oben definiert) ausgewählt.| This message should be displayed if `Time Pass` is selected.
If there are multiple Subscriptions available, the first option (as defined above) will be selected by default. |Wenn mehrere Abonnements verfügbar sind, wird automatisch die erste Option (wie oben definiert) ausgewählt. | This message should be displayed if `Subscription` is selected.

# Mockup & Prototype
[Prototype here](https://sketch.cloud/s/45evm/a/lQ4ked/play)

![image](https://user-images.githubusercontent.com/41650164/60124579-8587f080-974f-11e9-9ca4-1aac469499be.png)

# Original Feature Request
https://laterpay.atlassian.net/browse/PRODUCT-327

### Problem Statement Description
Our pre-made integrations don't allow merchants to specify the order of items in the price list in the purchase overlay. Currently in Connector, we default to "whatever order the API returns these purchase options" and we don't give Merchants control. This often works well, but sometimes produces impractical results.

Merchants want to be able to sort the purchase option list in different ways. (ascending price, descending price, custom sort order based on index integer). Merchants also want control over which item is pre-selected as the default option. Currently we default that to one of the Single Purchases on the page.

**To summarise:**
- purchase option list order presets (asc/desc by price, possibly other?)
- purchase option list order custom (index number / drag and drop / etc)
- purchase option preselection (currently topmost item in list, which does not always make sense, might perform worse than e.g. preselecting most expensive subscription option)

### Example of affected page:
https://www.kevelaerer-blatt.de/sport-ist-nicht-mord/ (WordPress)
![image](https://user-images.githubusercontent.com/41650164/60124787-05ae5600-9750-11e9-9cef-a712226d15b4.png)

**Comment by thrijith** [comment]
@ksmoore125 Need translations for following strings too.

| English             |
|------------------------------------------------------------|
| Pricing options displayed in **default order**.                |
| Pricing options displayed in **ascending order**.              |
| Pricing options displayed in **descending order**.             |
| **First option** is selected by default.                       |
| **Single Purchase** is selected by default.                    |
| **Time Pass** is selected by default.                          |
| **Subscription** is selected by default.                       |
| * are currently available to be sold with LaterPay         |
| **No Items!** are currently available to be sold with LaterPay |
| Customize Overlay (optional)                               |

**Comment by ksmoore125** [comment]
English | German
-- | --
Pricing options displayed in default order. | Die Preisoptionen werden in der Standardreihenfolge angezeigt.
Pricing options displayed in ascending order. | Die Preisoptionen werden in aufsteigender Reihenfolge angezeigt.
Pricing options displayed in descending order. | Die Preisoptionen werden in absteigender Reihenfolge angezeigt.
First option is selected by default. | Die erste Option ist standardmäßig ausgewählt.
Single Purchase is selected by default. | Einzelkauf ist standardmäßig ausgewählt. 
Time Pass is selected by default. | Time Pass ist standardmäßig ausgewählt.
Subscription is selected by default. | Das Abonnement ist standardmäßig ausgewählt.
* are currently available to be sold with LaterPay | * sind derzeit verfügbar, um mit LaterPay verkauft zu werden.
No Items! are currently available to be sold with LaterPay | Keine Artikel! sind derzeit verfügbar, um mit LaterPay verkauft zu werden.
Customize Overlay (optional) | Overlay anpassen (optional)

**Comment by ksmoore125** [comment]
@thrijith - The functionality looks great. Please make the following minor updates:

- Please make all of the options under `In what order should the purchase options be displayed?` plain (non-bold), black text so that it is consistent with the options under `Which purchase option should be selected by default?`
- Please update the first option under `In what order should the purchase options be displayed?` as follows:

English | German
-- | --
Default Order - Single Purchase followed by Time Passes and Subscriptions in the order they were created. | Standardbestellung - Einzelbestellung, gefolgt von Zeitpässen und Abonnements in der Reihenfolge ihrer Erstellung.

**Comment by thrijith** [comment]
@ksmoore125 Updated as suggested above.

**Comment by thrijith** [comment]
Released in [v2.6.1](https://github.com/laterpay/laterpay-wordpress-plugin/releases/tag/2.6.1)
